DDG4/src/Geant4SensDetAction.cpp

```diff
@@ -260,11 +263,16 @@ long long int Geant4Sensitive::volumeID(const G4VTouchable* touchable) {
 long long int Geant4Sensitive::cellID(const G4Step* step) {
   Geant4StepHandler h(step);
   Geant4VolumeManager volMgr = Geant4Mapping::instance().volumeManager();
-  VolumeID volID = volMgr.volumeID(h.preTouchable());
+  bool OpticsInvokeSD = G4OpticalParameters::Instance()?
+                        G4OpticalParameters::Instance()->GetBoundaryInvokeSD() : false;
+  bool IsOpticalPhoton = step->GetTrack()->GetDefinition() == G4OpticalPhoton::Definition();
+  bool UsePostStepOnly = IsOpticalPhoton && OpticsInvokeSD;
```

Comment:
```suggestion
  bool UsePostStepOnly = G4OpticalParameters::Instance() && G4OpticalParameters::Instance()->GetBoundaryInvokeSD() && (step->GetTrack()->GetDefinition() == G4OpticalPhoton::Definition());
```
If you are worried about performance?